torchrl/data/replay_buffers/storages.py

```diff
@@ -144,12 +144,19 @@ def load_state_dict(self, state_dict: Dict[str, Any]) -> None:
     def _empty(self):
         ...
 
+    # NOTE: This property is used to enable compiled Storages. A `len(self)`
+    # call can cause recompiles, but for some reason, wrapping the call in a
+    # `property` decorated function avoids the recompiles.
+    @property
```

Comment:
It seems that the `property` decorator does something to prevent the recompiles. If I remove the decorator and change `self.len` to `self.len()` below, the recompiles happen again. I'm wondering if this is either some kind of bug in the compiler, or if the compiler treats properties in a special way. Maybe it forces a graph break, forcing `len` to be executed outside of compiled code? I'll look through the verbose logs to try to find an answer

Reply:
That's weird! properties should not be treated differently than any other method I guess...

Reply:
Actually, it looks like the root cause really is the `mp.Value` after all! With the current state of this PR, I see this in the logs:

```
V1023 15:21:38.467000 663811 site-packages/torch/_dynamo/convert_frame.py:1182] [27/0] [__graph_breaks] torch._dynamo.exc.Unsupported: Graph break due to unsupported builtin None.SemLock.acq
uire. This function is either a Python builtin (e.g. _warnings.warn) or a third-party C/C++ Python extension (perhaps created with pybind). If it is a Python builtin, please file an issue on
 GitHub so the PyTorch team can add support for it and see the next case for a workaround. If it is a third-party C/C++ Python extension, please either wrap it into a PyTorch-understood cust
om operator (see https://pytorch.org/tutorials/advanced/custom_ops_landing_page.html for more details) or, if it is traceable, use torch.compiler.allow_in_graph.
V1023 15:21:38.467000 663811 site-packages/torch/_dynamo/convert_frame.py:1182] [27/0] [__graph_breaks] 
V1023 15:21:38.467000 663811 site-packages/torch/_dynamo/convert_frame.py:1182] [27/0] [__graph_breaks] from user code:
V1023 15:21:38.467000 663811 site-packages/torch/_dynamo/convert_frame.py:1182] [27/0] [__graph_breaks]    File "/home/endoplasm/develop/torchrl-0/torchrl/data/replay_buffers/storages.py", l
ine 159, in _rand_given_ndim
V1023 15:21:38.467000 663811 site-packages/torch/_dynamo/convert_frame.py:1182] [27/0] [__graph_breaks]     self.len,
V1023 15:21:38.467000 663811 site-packages/torch/_dynamo/convert_frame.py:1182] [27/0] [__graph_breaks]   File "/home/endoplasm/develop/torchrl-0/torchrl/data/replay_buffers/storages.py", li
ne 152, in len
V1023 15:21:38.467000 663811 site-packages/torch/_dynamo/convert_frame.py:1182] [27/0] [__graph_breaks]     return len(self)
V1023 15:21:38.467000 663811 site-packages/torch/_dynamo/convert_frame.py:1182] [27/0] [__graph_breaks]   File "/home/endoplasm/develop/torchrl-0/torchrl/data/replay_buffers/storages.py", li
ne 812, in __len__
V1023 15:21:38.467000 663811 site-packages/torch/_dynamo/convert_frame.py:1182] [27/0] [__graph_breaks]     return self._len
V1023 15:21:38.467000 663811 site-packages/torch/_dynamo/convert_frame.py:1182] [27/0] [__graph_breaks]   File "/home/endoplasm/develop/torchrl-0/torchrl/data/replay_buffers/storages.py", li
ne 469, in _len
V1023 15:21:38.467000 663811 site-packages/torch/_dynamo/convert_frame.py:1182] [27/0] [__graph_breaks]     return _len_value.value
V1023 15:21:38.467000 663811 site-packages/torch/_dynamo/convert_frame.py:1182] [27/0] [__graph_breaks]   File "<string>", line 3, in getvalue
```

Still, I don't understand what the difference is between decorator versus no decorator. If I change the `self.len` back to `len(self)`, I no longer see a graph break at that point in the logs. I wonder if it could have something to do with the fact that `Storage.__len__` is an abstract method...